modules/gapi/samples/bind_after_deserialization_example.cpp

```diff
@@ -0,0 +1,18 @@
+#include <opencv2/gapi.hpp>
+#include <opencv2/gapi/s11n.hpp>
+#include <opencv2/gapi/garg.hpp>
+
+int main()
+{
```

Comment:
> modules/gapi/samples

It makes sense to put these documentation-only non-functional samples somewhere else.

Reply:
@alalek I believe `gapi/samples` is the only place for such things - it provides examples for docs (via `@snippet`) and also checks that the code compiles

Reply:
OpenCV 'core' snippets are stored into `<opencv>/samples/cpp/tutorial_code/core/...`

Users expecting in `gapi/samples` full-functional samples which can be run and show some meanungful result.

---

> examples for docs

Check generated Doxyfile - there are many paths for storing that.

---

> `@snippet custom_type_serialization.cpp`

This filename is not enough to be unique across whole project (other cases have similar problem).
Use path with `gapi` entry (append path prefix) to avoid ambiguous problem in the future.

Reply:
@alalek I fixed the prefixes where applicable. Speaking of an appropriate place for these snippets: I've found 2 incomplete samples in g-api: `dynamic_graph.cpp` and `kernel_api_snippets.cpp`. I've copied `std::cout << "This sample is non-complete. It is used as code snippents in documentation." << std::endl;` to my samples. Is that enough? If you think that's still an issue can it be fixed separately?

Reply:
Also, can we merge this without a second Dmitry's review?

Reply:
> It makes sense to put these documentation-only non-functional samples somewhere else.

@alalek , Our documentation-related code (so it is "snippets) was there from the early beginning but if there's a better place, we can move it there for sure for all the snippets.